File: crud_java/src/main/java/com/example/crud_java/domain/item/ItemRepository.java
package com.example.crud_java.domain.item;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.time.LocalDateTime;
import java.util.Optional;

public interface ItemRepository {
    Page<Item> findAllCustom(Pageable pageable, String keyword, LocalDateTime startDate, LocalDateTime endDate);
    void save(Item item);
    Optional<Item> findById(Long itemNo);
}

File: crud_kotlin/src/main/kotlin/com/example/crud_kotlin/domain/item/ItemRepository.kt
package com.example.crud_kotlin.domain.item

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.time.LocalDateTime

interface ItemRepository {
    fun findAllCustom(
        pageable: Pageable,
        keyword: String?,
        startDate: LocalDateTime?,
        endDate: LocalDateTime?
    ): Page<Item>
    fun save(item: Item): Long
    fun saveAndFlush(item: Item): Item
    fun findByIdOrNull(itemNo: Long): Item?
}